judge/views/problem.py

```diff
@@ -201,6 +202,90 @@ def get_context_data(self, **kwargs):
                                           context['description'], 'problem')
         context['meta_description'] = self.object.summary or metadata[0]
         context['og_image'] = self.object.og_image or metadata[1]
+
+        context['vote_perm'] = self.object.get_vote_perm(user)
+        if context['vote_perm'].can_vote():
+            try:
+                context['vote'] = ProblemPointsVote.objects.get(voter=user.profile, problem=self.object)
+            except ObjectDoesNotExist:
+                context['vote'] = None
+        else:
+            context['vote'] = None
+
+        return context
+
+
+class ProblemVote(ProblemMixin, DetailView):
+    context_object_name = 'problem'
+    template_name = 'problem/vote-ajax.html'
+
+    def get_context_data(self, **kwargs):
+        if not self.object.get_vote_perm(self.request.user).can_vote():
+            raise Http404()
+
+        context = super().get_context_data(**kwargs)
+
+        try:
+            context['vote'] = ProblemPointsVote.objects.get(voter=self.request.profile, problem=self.object)
+        except ObjectDoesNotExist:
+            context['vote'] = None
+
+        context['max_possible_vote'] = settings.DMOJ_PROBLEM_MAX_USER_POINTS_VOTE
+        context['min_possible_vote'] = max(settings.DMOJ_PROBLEM_MIN_USER_POINTS_VOTE,
+                                           settings.DMOJ_PROBLEM_MIN_PROBLEM_POINTS)
+        return context
```

Comment:
If `DMOJ_PROBLEM_MIN_PROBLEM_POINTS` > `DMOJ_PROBLEM_MIN_USER_POINTS_VOTE` won't this unconditionally cause the model validation to fail?